owncloudApp/src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.kt

```diff
@@ -0,0 +1,2258 @@
+                    ), null
+                )
+
+                if (cursorOnKeptInSync != null && cursorOnKeptInSync.moveToFirst()) {
+                    var file: OCFile?
+                    var accountName: String
+                    do {
+                        file = createFileInstance(cursorOnKeptInSync)
+                        accountName = cursorOnKeptInSync.getString(
+                            cursorOnKeptInSync.getColumnIndex(ProviderTableMeta.FILE_ACCOUNT_OWNER)
+                        )
+                        if (!file!!.isFolder && AccountUtils.exists(accountName, mContext)) {
+                            result.add(Pair(file, accountName))
+                        }
+                    } while (cursorOnKeptInSync.moveToNext())
+                }
+
+            } catch (e: Exception) {
+                Log_OC.e(TAG, "Exception retrieving all the available offline files", e)
+
+            } finally {
+                cursorOnKeptInSync?.close()
+            }
+
+            return result
+        }
+
+    /**
+     * Get a collection with all the files set by the user as available offline, from current account
+     * putting away files whose parent is also available offline
+     *
+     * @return List with all the files set by current user as available offline.
+     */
+    // query for available offline files in current account and whose parent is not.
+    val availableOfflineFilesFromCurrentAccount: Vector<OCFile>
+        get() {
+            val result = Vector<OCFile>()
+
+            var cursorOnKeptInSync: Cursor? = null
+            try {
+                cursorOnKeptInSync = contentResolver!!.query(
+                    ProviderTableMeta.CONTENT_URI, null,
+                    "(" + ProviderTableMeta.FILE_KEEP_IN_SYNC + " = ? AND NOT " +
+                            ProviderTableMeta.FILE_KEEP_IN_SYNC + " = ? ) AND " +
+                            ProviderTableMeta.FILE_ACCOUNT_OWNER + " = ? ",
+                    arrayOf(
+                        OCFile.AvailableOfflineStatus.AVAILABLE_OFFLINE.value.toString(),
+                        OCFile.AvailableOfflineStatus.AVAILABLE_OFFLINE_PARENT.value.toString(),
+                        account!!.name
+                    ), null
+                )
+
+                if (cursorOnKeptInSync != null && cursorOnKeptInSync.moveToFirst()) {
+                    var file: OCFile?
+                    do {
+                        file = createFileInstance(cursorOnKeptInSync)
+                        result.add(file)
+                    } while (cursorOnKeptInSync.moveToNext())
+                }
+
+            } catch (e: Exception) {
+                Log_OC.e(TAG, "Exception retrieving all the available offline files", e)
+
+            } finally {
+                cursorOnKeptInSync?.close()
+            }
+
+            Collections.sort(result)
+            return result
```

Comment:
why don't you replace these 2 lines by

`return result.sort() `

Reply:
`return result.apply{ sort() }` 👍